test/framework/src/main/java/org/elasticsearch/common/util/MockBigArrays.java

```diff
@@ -37,12 +46,50 @@
 import java.util.Random;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
 
 import static org.elasticsearch.test.ESTestCase.assertBusy;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class MockBigArrays extends BigArrays {
+    private static final Logger logger = LogManager.getLogger(MockBigArrays.class);
+
+    /**
+     * Assert that a function returning a {@link Releasable} runs to completion
+     * when allocated a breaker with that breaks when it uses more than {@code max}
+     * bytes <strong>and</strong> that the function doesn't leak any
+     * {@linkplain BigArray}s if it is given a breaker that allows fewer bytes.
+     */
+    public static void assertFitsIn(ByteSizeValue max, Function<BigArrays, Releasable> run) {
+        long maxBytes = 0;
```

Comment:
great test, I am wondering if it would be better to have a specific class to test allocations instead of a static method here? The point is that it is difficult to find out which BigArrays implementation are actually tested. It can be done in a follow up PR or I might be wrong.

Reply:
I'm not sure! I didn't go with the superclass because I wasn't sure that every unit test where I make the method just extends `ESTestCase` now. They all do it right now but I was thinking "I'm going to want this in a bunch of spots." so I wasn't sure if it'd be safe to keep it in a superclass. So, I guess what I'm saying is, yeah, I think its a good idea but I'd prefer to do it in a follow up after I get a little more experience using this method.

Reply:
👍